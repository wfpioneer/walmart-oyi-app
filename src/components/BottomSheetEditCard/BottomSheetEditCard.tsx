import React from 'react';
import BottomSheetBaseCard from '../BottomSheetBaseCard/BottomSheetBaseCard';

interface EditCardProps {
  // eslint-disable-next-line react/require-default-props
  isVisible?: boolean;
  text: string;
  onPress: () => void;
}

const BottomSheetEditCard = (props: EditCardProps): JSX.Element | null => {
  const { isVisible, text, onPress } = props;

  return isVisible ? (
    <BottomSheetBaseCard
      image={require('../../assets/images/edit.png')}
      onPress={onPress}
      text={text}
    />
  ) : null;
};

export default BottomSheetEditCard;
