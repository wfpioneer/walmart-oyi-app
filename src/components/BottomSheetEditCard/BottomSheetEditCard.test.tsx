import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import BottomSheetEditCard from './BottomSheetEditCard';

describe('Edit item render tests', () => {
  it('renders the bottom sheet edit item, visible', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <BottomSheetEditCard
        onPress={() => {}}
        text="Edit an item"
        isVisible={true}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
