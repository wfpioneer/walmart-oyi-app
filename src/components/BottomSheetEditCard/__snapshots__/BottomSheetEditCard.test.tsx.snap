// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit item render tests renders the bottom sheet edit item, visible 1`] = `
<BottomSheetBaseCard
  image=""
  materialIconName=""
  onPress={[Function]}
  text="Edit an item"
/>
`;
